src/unit_tests.rs

```diff
@@ -12,6 +12,19 @@ fn test_creation() {
     expected.insert(&ONE, 1);
     assert!(counter.map == expected);
 }
+
+#[test]
+fn test_creation_with_capacity() {
+    let initializer = &[1];
+    let counter = Counter::init_with_capacity(initializer,1);
+
+    let mut expected = HashMap::with_capacity(1);
+    static ONE: usize = 1;
+    expected.insert(&ONE, 1);
+    assert!(counter.map == expected);
+    assert!(counter.map.capacity() == expected.capacity());
```

Comment:
We can simplify this.

```suggestion
    let counter = Counter::with_capacity(1);
    assert_eq!(counter.map.capacity(), 1);
```